kyo-core/shared/src/main/scala/kyo/Async.scala

```diff
@@ -179,13 +179,15 @@ object Async:
         def fromFuture[A: Flat](f: Future[A])(using Frame): Fiber[Nothing, A] < IO =
             import scala.util.*
             IO {
-                val p = new IOPromise[Nothing, A]()
-                f.onComplete {
-                    case Success(v) =>
-                        p.complete(Result.success(v))
-                    case Failure(ex) =>
-                        p.complete(Result.panic(ex))
-                }(ExecutionContext.parasitic)
+                val p = new IOPromise[Nothing, A] with (Try[A] => Unit):
```

Comment:
A few optimizations are just merging function interfaces with `IOPromise`. It avoids the function allocation in addition to the promise.